frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailWrapper.tsx

```diff
@@ -43,6 +43,10 @@ export function ObjectDetailWrapper({
 
   const hasPagination = data?.rows?.length > 1;
 
+  if (data.rows.length <= currentObjectIndex) {
```

Comment:
:x: I think that when query changes, it makes more sense to reset pagination state altogether (i.e. user should end up on the first page) - as per issue's original title: "Object detail navigation state is not reset (...)".